src/inspector_agent.cc

```diff
@@ -465,10 +465,17 @@ class NodeInspectorClient : public V8InspectorClient {
   }
 
   void waitForIoShutdown() {
+    CHECK(is_main_);
     waiting_for_io_shutdown_ = true;
     runMessageLoop();
   }
 
+  void workerWaitForDisconnect() {
```

Comment:
I feel like this is really similar to a `waiting_for_io_shutdown_`. Maybe make them one?

Reply:
I did it this way in first iteration but it looks a little be confusing and definitely makes code much less understandable.

Reply:
What if you call it `waiting_for_sessions_disconnect_`?

Reply:
done!